src/main/java/bookapp/dto/user/UserLoginRequestDto.java

```diff
@@ -0,0 +1,16 @@
+package bookapp.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import org.hibernate.validator.constraints.Length;
+
+public record UserLoginRequestDto(
+        @NotEmpty
+        @Length(min = 8, max = 20)
+        @Email
+        String email,
+        @NotEmpty
```

Comment:
the same. Let’s  read about the difference  https://www.baeldung.com/java-bean-validation-not-null-empty-blank